Makefile

```diff
@@ -41,7 +41,12 @@ generate-bsd-licenses:
 test:
 	go build -o artifacts/polygon-edge .
 	$(eval export PATH=$(shell pwd)/artifacts:$(PATH))
-	go test -timeout 28m ./...
+	go test -timeout 28m `go list ./... | grep -v e2e`
+
+.PHONY: test-e2e
+test-e2e:
+	go build -o polygon-edge main.go
```

Comment:
Are we testing for race conditions in E2E as well?

Reply:
Does it makes sense to test for race conditions when testing a binary?

Reply:
In our current E2E tests, the binary is built with a `-race` flag and the tests should fail on race condition: https://github.com/0xPolygon/polygon-edge/blob/49b9e68a71252a06a99f05b7f572ca4c5d22e894/.github/workflows/main.yml#L39-L51   

But, maybe, we can have a dedicated test that checks for race conditions? 
What do you @Kourin1996 @zivkovicmilos  guys think ?

Reply:
We can build with `-race` in the separated e2e tests too.

I've a question: why are you adding both CGO flag and GOOS in the action build?

Reply:
This should address your concern @ZeljkoBenovic https://github.com/0xPolygon/polygon-edge/pull/769/commits/2107135c9ea2b1e3fec30529ca5f53d4f2cfa947